components/user/UserPage.tsx

```diff
@@ -1,26 +1,61 @@
 import { useEffect, useState } from "react";
-import { DBResponse } from "../../entities/IDBResponse";
-import { Owner } from "../../entities/IOwner";
+import { Owner, OwnerLite } from "../../entities/IOwner";
 import { useRouter } from "next/router";
-import { areEqualAddresses } from "../../helpers/Helpers";
 import { ConsolidatedTDHMetrics, TDHMetrics } from "../../entities/ITDH";
-import { fetchAllPages, fetchUrl } from "../../services/6529api";
+import { fetchAllPages } from "../../services/6529api";
 import UserPageDetails from "./details/UserPageDetails";
 import { IProfileAndConsolidations } from "../../entities/IProfile";
 import UserPageHeader from "./user-page-header/UserPageHeader";
-import UserPageFetching from "./UserPageFetching";
 import { commonApiFetch } from "../../services/api/common-api";
+import { GRADIENT_CONTRACT, MEMES_CONTRACT } from "../../constants";
+import { NFT, NFTLite } from "../../entities/INFT";
+import { Season } from "../../entities/ISeason";
 
 interface Props {
   user: string;
   connectedWallets: string[];
   profile: IProfileAndConsolidations;
+  mainAddress: string;
+  consolidatedTDH: ConsolidatedTDHMetrics | null;
+  memesLite: NFTLite[];
+  gradients: NFT[];
+  seasons: Season[];
 }
 
 export default function UserPage(props: Props) {
   const router = useRouter();
   const isConsolidation = props.profile.consolidation.wallets.length > 1;
 
+  const getConsolidatedOwned = (): OwnerLite[] => {
+    if (!props.consolidatedTDH) {
+      return [];
+    }
+    const cards: OwnerLite[] = [];
+    if (Array.isArray(props.consolidatedTDH.memes)) {
+      for (const meme of props.consolidatedTDH.memes) {
+        cards.push({
+          token_id: meme.id,
+          balance: meme.balance,
+          contract: MEMES_CONTRACT,
+        });
+      }
+    }
+
+    if (Array.isArray(props.consolidatedTDH.gradients)) {
+      for (const gradient of props.consolidatedTDH.gradients) {
+        cards.push({
+          token_id: gradient.id,
+          balance: gradient.balance,
+          contract: GRADIENT_CONTRACT,
+        });
+      }
+    }
+
+    return cards;
+  };
+
+  const consolidatedOwned = getConsolidatedOwned();
```

Comment:
if you buy a card now
it will not be in this `consolidatedTDH.memes` or `consolidatedTDH.gradients` array
those contain data as per last tdh calculation

Reply:
Changed to old approach